src/Products/Annotation/Util/AnnotationMapper.cs

```diff
@@ -48,34 +53,67 @@ public AnnotationDataEntity[] mapForPage(AnnotationInfo[] annotations, int pageN
+                boxY = box.Y;
+                boxHeight = box.Height;
+                boxWidth = box.Width;
+
+                StringBuilder builder = new StringBuilder().
+                Append("M").Append(box.X.ToString(CultureInfo.InvariantCulture)).
+                Append(",").Append(box.Y.ToString(CultureInfo.InvariantCulture)).
+                Append("L").Append(box.Width.ToString(CultureInfo.InvariantCulture)).
+                Append(",").Append(box.Height.ToString(CultureInfo.InvariantCulture));
+
+                svgPath = builder.ToString();
+            }
+
             AnnotationDataEntity annotation = new AnnotationDataEntity();
-            annotation.font = annotationInfo.FontFamily;
-            double fontSize = Convert.ToDouble((annotationInfo.FontSize == null) ? 0 : annotationInfo.FontSize);
+            annotation.font = annotationInfo is IFontFamily ? ((IFontFamily)annotationInfo).FontFamily : "";
+            double fontSize = annotationInfo is IFontSize ? Convert.ToDouble((((IFontSize)annotationInfo).FontSize == null) ? 0 : ((IFontSize)annotationInfo).FontSize) : 0;
             annotation.fontSize = (float)fontSize;
-            annotation.fontColor = (annotationInfo.FontColor == null) ? 0 : (int)annotationInfo.FontColor;
-            annotation.height = annotationInfo.Box.Height;
-            annotation.left = annotationInfo.Box.X;
-            annotation.pageNumber = (int)annotationInfo.PageNumber + 1;           
-            annotation.svgPath = (annotationInfo.SvgPath != null) ? annotationInfo.SvgPath.Replace("l", "L") : null;
-            string text = (annotationInfo.Text == null) ? annotationInfo.FieldText : annotationInfo.Text;
+            annotation.fontColor = annotationInfo is IFontColor ? ((((IFontColor)annotationInfo).FontColor == null) ? 0 : (int)((IFontColor)annotationInfo).FontColor) : 0;
+            annotation.height = annotationInfo is IBox ? boxHeight : (annotationInfo is IPoints ? (maxY - minY) : 0);
+            annotation.left = annotationInfo is IBox ? boxX : (annotationInfo is IPoints ? minX : 0);
+            annotation.pageNumber = (int)annotationInfo.PageNumber + 1;
+            annotation.svgPath = annotationInfo is ISvgPath ? (((ISvgPath)annotationInfo).SvgPath?.Replace("l", "L")) : svgPath;
+            string text = annotationInfo is IText ? (((IText)annotationInfo).Text ?? (annotationInfo is ITextToReplace ? ((ITextToReplace)annotationInfo).TextToReplace : "")) : "";
             annotation.text = text;
-            annotation.top = annotationInfo.Box.Y;
-            annotation.type = Char.ToLowerInvariant(Enum.GetName(typeof(AnnotationType), annotationInfo.Type)[0]) + Enum.GetName(typeof(AnnotationType), annotationInfo.Type).Substring(1);
-            annotation.width = annotationInfo.Box.Width;
+            // TODO: implement backward top fix only for specific annotation types
+            //annotation.top = annotationInfo is IBox ? boxY : (annotationInfo is IPoints ? minY : 0);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Annotation-for-.NET-MVC/pullRequest?prid=5372470)